### DISCUSSION TITLE
How to modify a .txt file

### DISCUSSION THREAD

**Discussion by jantje** [Sloeber/arduino-eclipse-plugin#1533] [Show and tell]
How to modify a .txt file

    1) Modify the .txt file
    2) delete the corresponding *.sloeber.txt file.
    3) for each project that should be impacted: goto project properties->sloeber->apply and close
    3 check) if the sloeber.txt file is not there something went wrong and your change has not been taken into account in any project by Sloeber.
    4) force Sloeber to recreate the makefile by deleting the build output folder
    5) build

Note 
1) that if you do not do 3 or 4 for each and every project some project will use the old settings and others the new settings which will confuse you in the future.
So do them all and test whether all have been done properly.
2) If you upgrade your product your changes will be lost. Therefore the following note may contain a better way.
3) As an alternative to modifying .txt files; **you can overwrite .txt provided settings in sloeber** at project properties->sloeber->compile options. The text provided there will be added at the end of the command and GCC convention is that later provided parameters overrule earlier provided parameters.

**Comment by gfvalvo** [comment]
@GitHubLionel,
The instructions are kind of confusing and they could have been worded better. But they do work. Here’s what I did:

1. Go to the file containing ‘platform.txt’. The path will be something like ‘../../arduinoPlugin\packages\esp32\hardware\esp32\2.0.11’

2. Delete the file ‘platform.sloeber.txt’.

3. Open the file ‘platform.txt’ with a text editor.

4. Add these lines to the end:
recipe.hooks.prebuild.7.pattern=bash -c "[ -f "{build.path}"/file_opts ] || touch "{build.path}"/file_opts"
recipe.hooks.prebuild.7.pattern.windows=cmd /c if not exist "{build.path}\file_opts" type nul > "{build.path}\file_opts"

5. Save and exit the editor.

6. Open Sloeber. Then open each of your ESP32 projects that use this core. And go to: ‘properties->sloeber->apply and close’.

There’s also something in the instructions about deleting the “build output folder”. But I don’t understand what that means since there is no folder called “build”.

I hope that this bug will be fixed in a future release of Sloeber. But, until then, the workaround hack does the job.

**Reply by GitHubLionel** [reply]
Hi gfvalvo

Many thanks, work perfectly :)

Best regards,
Lionel

**Reply by PerennialNovice** [reply]
the "build output folder" usually is called "Release" - at least with latest sloeber and Arduino2560 as well as STM32 Nucleo boards.
If you delete this, it is like a brute force "build -clean" since all build output is deleted as well as all generated build files and data.

**Comment by PerennialNovice** [comment]
@jantje , is there a way to append stuff to the upload process similar to your note 3 ?
I would like to get rid of the verify process from avrdude without having to fiddle with the .txt files. I doubt the change there will stay in case of a platform update ...

**Reply by jantje** [reply]
> is there a way to append stuff to the upload process

I currently see no other way than modifying .txt files or add a makefile.targets to you project and add the upload commands and use the CDT run target to upload.
 
> I doubt the change there will stay in case of a platform update ...

If it did it would be a bug
